File: demo-test/src/test/java/io/fries/demo/test/cucumber/steps/ParameterTypes.java
package io.fries.demo.test.cucumber.steps;

import io.cucumber.java.ParameterType;
import io.fries.demo.test.cucumber.world.World;
import io.fries.demo.test.cucumber.world.request.QueryParameter;
import org.springframework.beans.factory.annotation.Autowired;

import java.time.LocalTime;
import java.time.ZonedDateTime;

public class ParameterTypes {

    @Autowired
    private World world;

    @ParameterType("(.*)=(.*)")
    public QueryParameter queryParameter(final String name, final String value) {
        return new QueryParameter(name, value);
    }

    @ParameterType("\\d{2}:\\d{2}")
    public ZonedDateTime zonedDateTime(final String time) {
        return world.clock()
                .now()
                .with(LocalTime.parse(time));
    }
}

File: demo-test/src/test/kotlin/io/fries/demo/test/cucumber/steps/ParameterTypes.kt
package io.fries.demo.test.cucumber.steps

import io.cucumber.java.ParameterType
import io.fries.demo.test.cucumber.world.World
import io.fries.demo.test.cucumber.world.request.QueryParameter
import java.time.LocalTime
import java.time.ZonedDateTime

class ParameterTypes(private val world: World) {

    @ParameterType("(.*)=(.*)")
    fun queryParameter(name: String, value: String): QueryParameter = QueryParameter(name, value)

    @ParameterType("\\d{2}:\\d{2}")
    fun zonedDateTime(time: String): ZonedDateTime = world.clock()
        .now()
        .with(LocalTime.parse(time))
}